src/main/java/org/walkmod/javalang/compiler/types/TypesLoaderVisitor.java

```diff
@@ -135,22 +135,27 @@ public TypesLoaderVisitor(SymbolTable symbolTable, SymbolActionProvider actionPr
     * @param imported
     *           if it appears as an import declaration. It is important, because the simple name is
     *           the part after the $ symbol if it is an inner class.
+    * @param importedName
+    *           the name of the import to resolve nested classes against (optional)
     * @return the simple name to be pushed into the symbol table
     */
-   private String getSymbolName(String name, boolean imported) {
+   //@VisibleForTesting
+   static String registrationSymbolName(final String name, final boolean imported, String importedName) {
       int index = name.lastIndexOf(".");
       String simpleName = name;
       if (index != -1) {
          simpleName = name.substring(index + 1);
       }
-      if (imported) {
-         index = simpleName.lastIndexOf("$");
-         if (index != -1) {
-            simpleName = simpleName.substring(index + 1);
+      if (importedName != null && name.startsWith(importedName + "$")) {
```

Comment:
It is OK for me if the existing tests are working. However, what I would prefer is to have a boolean `isImportedNestedClass` instead of `importedName`